src/ty/mod.rs

```diff
@@ -405,6 +397,49 @@ where
     }
 }
 
+/// A type wrapped in [`Compact`].
+#[derive(PartialEq, Eq, PartialOrd, Ord, Clone, Encode, Decode, Debug)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+#[cfg_attr(
+    feature = "serde",
+    serde(bound(
+        // TODO: check these bounds
```

Comment:
Actually I think they are unnecessary because `T::Type` is used directly as a field type and `serde` must add the bounds automatically in that case. Just made https://github.com/paritytech/scale-info/pull/60 which removes other redundant ones.

Reply:
Nice, good riddance.